File: unity-samples/unity-app/Assets/Scripts/RoomSpec.cs
using System;
using System.Collections.Generic;
using UnityEngine;

public class RoomSpec
{
    public enum Type
    {
        SFU,
        SFU_LARGE,
        P2P,
        P2P_TURN,
    }

    private Type type;
    public RoomSpec(Type type)
    {
        this.type = type;
    }

    public AndroidJavaObject GetSpecHashMap()
    {
        var mediaControl = new Dictionary<string, object>
        {
            ["bitrate_reservation_mbps"] = 25
        };

        var dic = new Dictionary<string, object>
        {
            ["type"] = Enum.GetName(typeof(Type), type).ToLower(),
            ["media_control"] = mediaControl
        };
        return Utils.GetHashMap(dic);
    }
}

File: android-device-samples/theta-plugin/src/main/java/com/ricoh/livestreaming/theta/RoomSpec.kt
/*
 * Copyright 2022 RICOH Company, Ltd. All rights reserved.
 */

package com.ricoh.livestreaming.theta

class RoomSpec(
        private val roomType: RoomType
) {

    enum class RoomType(val typeStr: String) {
        SFU("sfu"),
        SFU_LARGE("sfu_large"),
        P2P("p2p"),
        P2P_TURN("p2p_turn")
    }

    fun getSpec(): Map<String, Any> {
        val map = HashMap<String, Any>()
        map["type"] = roomType.typeStr
        val mediaControl = HashMap<String, Any>()
        mediaControl["bitrate_reservation_mbps"] = 25
        map["media_control"] = mediaControl

        return map
    }
}
